### PR TITLE
merging coherences of boi and soi

### PR DESCRIPTION
**Pull request by mnergizci**

### CODE REVIEW

File: bin/scripts/create_bovl_ifg.sh [hunk 1]
```diff
@@ -140,8 +140,9 @@ if [ -f GEOC/$pair/$pair.geo.bovldiff.$tadf'tif' ]; then
 fi
 if [ -f GEOC/$pair/$pair.geo.bovldiff.$tadf'cc.tif' ]; then
  mv GEOC/$pair/$pair.geo.bovldiff.$tadf'cc.tif' GEOC/$pair/$pair.geo.bovldiff.$tadf'cc.temp.tif';
- gdal_translate -ot Byte -scale 0 1 0 255 -co COMPRESS=DEFLATE -co PREDICTOR=2 -of GTiff -a_srs EPSG:4326 GEOC/$pair/$pair.geo.bovldiff.$tadf'cc.temp.tif' GEOC/$pair/$pair.geo.bovldiff.$tadf'cc.tif'
```
Comment (espiritocz) [inline bin/scripts/create_bovl_ifg.sh:143]: We should store coh in 8-bit. Please return this line back and update your procedure to load 8-bit data

File: python/create_sbovl_ifg.py [hunk 2]
```diff
@@ -38,25 +38,57 @@
...
 if os.path.exists(soi_path):
     soi = open_geotiff(soi_path, fill_value=np.nan)
 else:
     print(f"{soi_path} doesn't exist ,please run create_soi.py first!")
     sys.exit(1)
-    
+
+# Open coherences
+if os.path.exists(boi_coh_path):
+    boi_coh = open_geotiff(boi_coh_path, fill_value=np.nan)
```
Comment (espiritocz) [inline python/create_sbovl_ifg.py:61]: not sure what fill_value will do - boi_coh is uint8. consider 0->np.nan. You can do just
boi_coh = boi_coh/255
boi_coh[boi_coh==0] = np.nan

File: python/create_sbovl_ifg.py [hunk 3]
```diff
@@ -38,25 +38,57 @@
...
+
+# Process and scale coherence data
+super_sboi_coh = boi_coh.copy()
+super_sboi_coh[super_sboi_coh==0] = np.nan
+soi_coh[soi_coh==0]=np.nan
+super_sboi_coh[np.isnan(super_sboi_coh)] = soi_coh[np.isnan(super_sboi_coh)]
+
+# Scale coherence data to 0-255
+super_sboi_coh[super_sboi_coh==0] = np.nan 
+super_sboi_coh = np.clip(super_sboi_coh * 255, 0, 255)
```
Comment (espiritocz) [inline python/create_sbovl_ifg.py:87]: not clear to me (max coh should be 1, i.e. 255. why need to clip above?)

File: python/create_sbovl_ifg.py [hunk 4]
```diff
@@ -38,25 +38,57 @@
...
-super_sboi[np.isnan(super_sboi)] = np.nan
+
+# Process and scale coherence data
+super_sboi_coh = boi_coh.copy()
+super_sboi_coh[super_sboi_coh==0] = np.nan
+soi_coh[soi_coh==0]=np.nan
+super_sboi_coh[np.isnan(super_sboi_coh)] = soi_coh[np.isnan(super_sboi_coh)]
+
+# Scale coherence data to 0-255
+super_sboi_coh[super_sboi_coh==0] = np.nan
```
Comment (espiritocz) [inline python/create_sbovl_ifg.py:86]: better to do opposite:
super_sboi_coh[np.isnan(super_sboi_coh)] = 0
and then (removing line 87) export for example to float tif, and then use os.system to just convert through gdal:
cmd = 'gdal_translate -ot Byte -scale 0 1 0 255 -co COMPRESS=DEFLATE -co PREDICTOR=2 -of GTiff -a_srs EPSG:4326 .................'
os.system(cmd)
(after line 92)

**Review by espiritocz — CHANGES_REQUESTED** [review]
hi, please update

File: bin/scripts/create_bovl_ifg.sh (hunk 1 above)
Comment (mnergizci) [reply bin/scripts/create_bovl_ifg.sh:143]: If we do save the coh as 8-bit, the problem arises in merging step in create_sbovl.py, the nan values can't be easily handled like in float32 and I get 'Error of cannot convert float NaN to integer '?

Comment (mnergizci) [reply bin/scripts/create_bovl_ifg.sh:143]: I see the other comments, let me try all of them and inform you if works fine

File: python/create_sbovl_ifg.py (hunk 3 above)
Comment (mnergizci) [reply python/create_sbovl_ifg.py:87]: np.clip is a function that limits the values in an array to a spefied range:
np.clip(array, min_value, max_value)
Any value in the array that falls outside this range is replaced by the boundary values so I used that for double checking  after multiply the 0-1 values with 255, but It is redundant if the all values fall in between 0-255:
super_sboi_coh = np.clip(super_sboi_coh * 255, 0, 255)

**Pull request closed without merging**